File: Leetcode/Python/_242.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s = self.dict(s)
        t = self.dict(t)
        if len(s) != len(t):
            return False
        else:
            for key, val in s.items():
                value = t.get(key, False)
                if value:
                    if value != val:
                        return False
                else:
                    return False
        return True
        
        
    def dict(self, s):
        hashMap = {}
        for string in s:
            if string not in hashMap:
                hashMap[string] = 0
            hashMap[string] +=1
        return hashMap 

File: Leetcode/cpp/242.cpp
class SolutionV2 {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) return false;
        std::sort(s.begin(), s.end());
        std::sort(t.begin(), t.end());
        for (int i = 0; i < s.length(); ++i) {
            if (s[i] != t[i]) return false;
        }
        return true;
    }
};

class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) {
            return false;
        }
        std::vector<int> map(26, 0);
        for (int i = 0; i < t.length(); i++) {
            map[s[i] - 'a']++;
            map[t[i] - 'a']--;
        }
        for (int i = 0; i < 26; i++) {
            if (map[i] != 0) {
                return false;
            }
        }
        return true;
    }
};
